### PR TITLE
ALTAPPS-800: Android replace placeholder screen with track selection

### PR DESCRIPTION
**Pull request by XanderZhu**
**YouTrack Issues**:
[#ALTAPPS-800](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-800)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [x] All checks have been passed;
- [x] Changes have been checked locally.

**Description**
- Remove Android platform dependencies for the PlaceholderNewUser screen
- Replace navigation to the PlaceholderNewUserScreen with navigation to the TrackSelectionListScreen

### CODE REVIEW

File: androidHyperskillApp/src/main/java/org/hyperskill/app/android/placeholder_new_user/fragment/PlaceholderNewUserFragment.kt [hunk 1]
```diff
@@ -1,186 +0,0 @@
...
-                requireRouter().newRootScreen(MainScreen)
-            }
-            PlaceholderNewUserFeature.Action.ViewAction.ShowTrackSelectionStatus.Loading -> {
-                LoadingProgressDialogFragment.newInstance()
-                    .showIfNotExists(childFragmentManager, LoadingProgressDialogFragment.TAG)
-            }
-            PlaceholderNewUserFeature.Action.ViewAction.ShowTrackSelectionStatus.Error -> {
-                childFragmentManager
-                    .dismissDialogFragmentIfExists(LoadingProgressDialogFragment.TAG)
-                viewBinding.root.snackbar(org.hyperskill.app.R.string.placeholder_new_user_start_track_error_message)
```
Comment (ivan-magda) [inline androidHyperskillApp/src/main/java/org/hyperskill/app/android/placeholder_new_user/fragment/PlaceholderNewUserFragment.kt:135]: Please check maybe we can delete this string resource `placeholder_new_user_start_track_error_message`

File: androidHyperskillApp/src/main/java/org/hyperskill/app/android/placeholder_new_user/fragment/PlaceholderNewUserFragment.kt [hunk 2]
```diff
@@ -1,186 +0,0 @@
...
-            }
-            PlaceholderNewUserFeature.Action.ViewAction.ShowTrackSelectionStatus.Success -> {
-                childFragmentManager
-                    .dismissDialogFragmentIfExists(LoadingProgressDialogFragment.TAG)
-                viewBinding.root.snackbar(org.hyperskill.app.R.string.placeholder_new_user_start_track_success_message)
```
Comment (ivan-magda) [inline androidHyperskillApp/src/main/java/org/hyperskill/app/android/placeholder_new_user/fragment/PlaceholderNewUserFragment.kt:140]: Please check maybe we can delete this string resource `placeholder_new_user_start_track_success_message`?

File: androidHyperskillApp/src/main/res/layout/fragment_placeholder_new_user_screen.xml [hunk 3]
```diff
@@ -1,123 +0,0 @@
...
-                android:id="@+id/placeholderRecyclerView"
-                android:layout_width="match_parent"
-                android:layout_height="0dp"
-                android:layout_weight="1"
-                android:nestedScrollingEnabled="false"
-                android:paddingTop="32dp"
-                android:clipToPadding="false"
-                app:layout_constraintTop_toBottomOf="@+id/placeholderInstructionTextView"
-                app:layout_constraintBottom_toBottomOf="parent"
-                tools:listitem="@layout/item_new_user_track"
```
Comment (ivan-magda) [inline androidHyperskillApp/src/main/res/layout/fragment_placeholder_new_user_screen.xml:71]: `layout/item_new_user_track` deleted?

File: androidHyperskillApp/src/main/java/org/hyperskill/app/android/placeholder_new_user/fragment/PlaceholderNewUserFragment.kt [hunk 4]
```diff
@@ -1,186 +0,0 @@
...
-import ru.nobird.android.ui.adapters.DefaultDelegateAdapter
-import ru.nobird.android.view.base.ui.delegate.ViewStateDelegate
-import ru.nobird.android.view.base.ui.extension.setTextIfChanged
-import ru.nobird.android.view.base.ui.extension.showIfNotExists
-import ru.nobird.android.view.base.ui.extension.snackbar
-import ru.nobird.android.view.redux.ui.extension.reduxViewModel
-import ru.nobird.app.presentation.redux.container.ReduxView
-
-class PlaceholderNewUserFragment :
-    Fragment(R.layout.fragment_placeholder_new_user_screen),
```
[the remark this answers is not in the document]
Comment (XanderZhu) [reply androidHyperskillApp/src/main/java/org/hyperskill/app/android/placeholder_new_user/fragment/PlaceholderNewUserFragment.kt:39]: Yes

File: androidHyperskillApp/src/main/java/org/hyperskill/app/android/placeholder_new_user/fragment/PlaceholderNewUserFragment.kt [hunk 5]
```diff
@@ -1,186 +0,0 @@
...
-                    }
-                    trackAdapter.items = viewData.tracks
-                }
-            }
-        }
-    }
-
-    private fun trackAdapterDelegate(imageLoader: ImageLoader) =
-        adapterDelegate<PlaceholderNewUserViewData.Track, PlaceholderNewUserViewData.Track>(
-            R.layout.item_new_user_track
```
[the remark this answers is not in the document]
Comment (XanderZhu) [reply androidHyperskillApp/src/main/java/org/hyperskill/app/android/placeholder_new_user/fragment/PlaceholderNewUserFragment.kt:167]: Yes

File: androidHyperskillApp/src/main/res/layout/fragment_placeholder_new_user_screen.xml [hunk 6]
```diff
@@ -1,123 +0,0 @@
...
-                tools:ignore="ContentDescription"
-                />
-
-            <TextView
-                android:id="@+id/placeholderRegisterTextView"
-                android:layout_width="match_parent"
-                android:layout_height="wrap_content"
-                android:layout_marginTop="20dp"
-                android:gravity="center_horizontal"
-                android:text="@string/placeholder_new_user_title"
```
[the remark this answers is not in the document]
Comment (XanderZhu) [reply androidHyperskillApp/src/main/res/layout/fragment_placeholder_new_user_screen.xml:41]: I guess, it is used on iOS side and should be deleted together with iOS code.

File: androidHyperskillApp/src/main/res/layout/fragment_placeholder_new_user_screen.xml [hunk 7]
```diff
@@ -1,123 +0,0 @@
...
-                app:layout_constraintStart_toStartOf="parent"
-                app:layout_constraintTop_toBottomOf="@+id/placeholderIcon" />
-
-            <TextView
-                android:id="@+id/placeholderInstructionTextView"
-                android:layout_width="match_parent"
-                android:layout_height="wrap_content"
-                android:layout_marginTop="32dp"
-                android:gravity="center_horizontal"
-                android:text="@string/placeholder_new_user_text"
```
[the remark this answers is not in the document]
Comment (XanderZhu) [reply androidHyperskillApp/src/main/res/layout/fragment_placeholder_new_user_screen.xml:54]: I guess, it is used on iOS side and should be deleted together with iOS code.

File: androidHyperskillApp/src/main/res/layout/fragment_placeholder_new_user_screen.xml (hunk 3 above)
Comment (XanderZhu) [reply androidHyperskillApp/src/main/res/layout/fragment_placeholder_new_user_screen.xml:71]: Yes

**Review by ivan-magda — APPROVED** [review]

**Pull request merged**